src/main/imageEvents.ts

```diff
@@ -5,160 +5,159 @@
 import path from 'path';
 
 import Electron from 'electron';
-
-import { ImageProcessor } from '@/k8s-engine/images/imageProcessor';
-import { createImageProcessor, ImageProcessorName } from '@/k8s-engine/images/imageFactory';
 import Logging from '@/utils/logging';
 import * as window from '@/window';
-import * as K8s from '@/k8s-engine/k8s';
+
+import { ImageProcessor } from '@/k8s-engine/images/imageProcessor';
 
 const console = Logging.images;
 
-let imageManager: ImageProcessor;
-let lastBuildDirectory = '';
-let mountCount = 0;
+interface ImageContents {
+  imageName: string,
+  tag: string,
+  imageID: string,
+  size: string
+}
 
-/**
- * Map image-related events to the associated image processor's methods
- * @param imageProcessorName
- * @param k8sManager
- */
+// TODO: Turn this into a singleton
+
+// Map image-related events to the associated image processor's methods
+// TODO: export the factory function to make this a singleton
 
-export function setupImageProcessor(imageProcessorName: ImageProcessorName, k8sManager: K8s.KubernetesBackend): ImageProcessor {
-  imageManager = imageManager ?? createImageProcessor(imageProcessorName, k8sManager);
+export class ImageEventHandler {
```

Comment:
We should have JSDoc describing why we might want to use this class.

Reply:
Added.